opencdc/toproto.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencdc
+
+import (
+	"fmt"
+
+	opencdcv1 "github.com/conduitio/conduit-commons/proto/opencdc/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func _() {
+	// An "invalid array index" compiler error signifies that the constant values have changed.
+	var cTypes [1]struct{}
+	_ = cTypes[int(OperationCreate)-int(opencdcv1.Operation_OPERATION_CREATE)]
+	_ = cTypes[int(OperationUpdate)-int(opencdcv1.Operation_OPERATION_UPDATE)]
+	_ = cTypes[int(OperationDelete)-int(opencdcv1.Operation_OPERATION_DELETE)]
+	_ = cTypes[int(OperationSnapshot)-int(opencdcv1.Operation_OPERATION_SNAPSHOT)]
+}
+
+func (r Record) ToProto(proto *opencdcv1.Record) error {
```

Comment:
The method FromProto is a pointer receiver method (as you also mentioned in Slack). Is there a special reason why this one isn't?

Reply:
Simply because it doesn't have to be 🙃 `Record` should still be treated as a value in most of the code, so by default methods should have a value receiver. `FromProto` and `UnmarshalJSON` are exceptions, as they mutate the record in place instead of returning a new record, so they get pointer receivers. The idea is that this will allow us to optimize allocations in Conduit, as we can then have a pool of records that we reuse over and over again, instead of allocating a new record each time.

Reply:
Gotcha, so we're consciously not being consistent (for the reasons you mentioned), that's what I wanted to confirm.